core/dbt/cli/flags.py

```diff
@@ -17,11 +17,28 @@
     # https://bugs.python.org/issue41567
     import multiprocessing.popen_spawn_posix  # type: ignore  # noqa: F401
 
+# TODO anything that has a default in params should be removed here?
+# Or maybe only the ones that's in the root click group
+flag_defaults = {
+    "SEND_ANONYMOUS_USAGE_STATS": True,
+    "INDIRECT_SELECTION": "eager",
+    "NO_PRINT": False,
+    "TARGET_PATH": None,
+    # cli args without user_config or env var option
+    "FULL_REFRESH": False,
+    "STRICT_MODE": False,
+    "STORE_FAILURES": False,
+}
+
 
 @dataclass(frozen=True)
 class Flags:
     def __init__(self, ctx: Context = None, user_config: UserConfig = None) -> None:
 
+        # set the default flags
+        for key, value in flag_defaults.items():
+            object.__setattr__(self, key, value)
```

Comment:
Yes, I was chatting with @MichelleArk  about similar thing also. @iknox-fa  might have something around it, but we are gonna defer that to work later on instead of in this PR

Reply:
I didn't do that the first time around because I wanted to keep things as immutable as python allows so adding even a "private" method to do that seemed like a recipe for mutated flags in the codebase in a few months.

Maybe as a compromise for readability we just make nested `get` and `set` functions in __init__?  That way no-one uses them unless they're directly editing the flags? We can discuss it further but given how much of a tangled mess the flags are currently I think we're going to get a lot of mileage out of having reasonable assurance of immutability and I don't want to jeopardize that.